#summary Frequently asked questions.
#labels Featured
= Introduction =

Below are common questions that are commonly asked by new BEDTools users. We have attempted to organize the content by subject.


== Problems with file formats ==
  * *What does zero-based, half-open mean?*
  BEDTools users are sometimes confused by the way BED start and end coordinates are represented. BEDTools uses the UCSC Genome Browser’s internal database convention of making the start position 0-based and the end position 1-based: (http://genome.ucsc.edu/FAQ/FAQtracks#tracks1). In other words, BEDTools interprets the “start” column as being 1 basepair higher than what is represented in the file. For example, the following BED feature represents a single base on chromosome 1; namely, the 1st base.
{{{
$ cat first_base.bed 
chr1	0	1	i-am-the-first-position-on-chrom1
}}}
  Why, you might ask? The advantage of storing features this way is that it prevents lots of superfluous "+1" operations in the code when computing lengths, overlap, etc. These would be necessary were the coordinate system 1-based, inclusive. Thus, storing BED features this way reduces the computational burden.

  * *Why must my input files be tab-delimited?*
  While it is certainly possible to allow BED and other formats to be delimited by any form of whitespace, we have decided to require tab delimiters for three reasons:
    # By requiring one delimiter type, the processing time is minimized. 
    # Tab-delimited files are more amenable to other UNIX utilities.
    # GFF files can contain spaces within attribute columns. This complicates the use of space-delimited files as spaces must therefore be treated specially depending on the context.
  

== Problems with output       == 




== Converting among formats   == 